src/Host/Client/Test/IdeGraphicsDeviceTest.cs

```diff
@@ -478,46 +501,22 @@ public class IdeGraphicsDeviceTest {
 ";
 
             var inputs = Interactive(code);
-            var actualPlotFilePaths = await GraphicsTestAgainstExpectedFilesAsync(inputs);
-            actualPlotFilePaths.Should().HaveCount(4);
-
-            var bmp1 = (Bitmap)Image.FromFile(actualPlotFilePaths[0]);
-            var bmp2 = (Bitmap)Image.FromFile(actualPlotFilePaths[1]);
-            var bmp3 = (Bitmap)Image.FromFile(actualPlotFilePaths[2]);
-            var bmp4 = (Bitmap)Image.FromFile(actualPlotFilePaths[3]);
-            bmp1.Width.Should().Be(DefaultWidth);
-            bmp1.Height.Should().Be(DefaultHeight);
-            bmp2.Width.Should().Be(DefaultWidth);
-            bmp2.Height.Should().Be(DefaultHeight);
-            bmp3.Width.Should().Be(600);
-            bmp3.Height.Should().Be(600);
-            bmp4.Width.Should().Be(600);
-            bmp4.Height.Should().Be(600);
-        }
-
-        private void CheckHistoryResult(string historyInfoFilePath, int expectedActive, int expectedCount) {
-            string json = File.ReadAllText(historyInfoFilePath).Trim();
-            json.Should().Be($"[{expectedActive},{expectedCount}]");
+            var actualPlotPaths = (await GraphicsTestAsync(inputs)).ToArray();
+            var expectedPlotPaths = new string[] { expected1Path, expected2Path, expected3Path, expected4Path };
+            CompareImages(actualPlotPaths, expectedPlotPaths);
         }
 
-        private async Task<string[]> GraphicsTestAgainstExpectedFilesAsync(string[] inputs) {
-            var actualPlotPaths = (await GraphicsTestAsync(inputs)).ToArray();
-            var expectedPlotPaths = GetTestExpectedFiles();
-            actualPlotPaths.Length.Should().Be(expectedPlotPaths.Length);
+        private void CompareImages(string[] actualPlotPaths, string[] expectedPlotPaths) {
+            actualPlotPaths.Should().HaveSameCount(expectedPlotPaths);
 
-            foreach (string path in expectedPlotPaths) {
-                var actualPlotPath = actualPlotPaths.First(p => Path.GetFileName(p) == Path.GetFileName(path));
-                File.ReadAllBytes(actualPlotPath).Should().Equal(File.ReadAllBytes(path));
+            for (int i = 0; i < actualPlotPaths.Length; i++) {
+                File.ReadAllBytes(actualPlotPaths[i]).Should().Equal(File.ReadAllBytes(expectedPlotPaths[i]));
```

Comment:
(although it might have to be `ShouldBeEquivalentTo` for it to compare data rather than references; `Be` will just use `Object.Equals`, I think, which doesn't compare data for arrays)